### DISCUSSION TITLE
Support for generating Javadoc documentation

### DISCUSSION THREAD

**Discussion by shalousun** [TongchengOpenSource/smart-doc#795] [Ideas]
# Description
Community users hope that smart-doc can generate simple Javadoc, such as documentation for some interface classes and static utility classes.
社区用户希望smart-doc能成生成简单的javadoc，例如针对一些接口类，静态工具类生成文档。 #683 
# Implementation
- Define a custom `@javadoc` annotation tag. Classes annotated with `@javadoc` will have a Javadoc document generated by smart-doc.
 定义一个自定义的`@javadoc`注释tag，标注了`@javadoc`的类，smart-doc会生成一个javadoc文档
- The generation of Javadoc is basically similar to the current Dubbo documentation. The Javadoc documentation needs to remove Protocol and URI attributes.
   实现javadoc的生成基本和当前dubbo的文档类似，javadoc文档需要去除Protocol、URI属性
- Add a `javadoc` mojo in the Maven plugin.
  在maven插件中增加一个javadoc的mojo

**Comment by tuichenchuxin** [comment]
I am very interested in this topic. Is there anything I can participate in to speed up the implementation of this function?

**Reply by shalousun** [reply]
Yes, we would very much welcome your participation and contributions.

**Reply by tuichenchuxin** [reply]
Okay, let me take a look at your implementation idea and do some code research.

**Comment by tuichenchuxin** [comment]
@shalousun 
Hello, I implemented a preliminary version according to your idea. If you have time, please help me to see if this idea is correct and how to merge this PR.
#797

**Reply by shalousun** [reply]
ok

**Comment by tuichenchuxin** [comment]
@shalousun Hello, I would like to ask when our version 3.0.5 will be released?

**Reply by shalousun** [reply]
> @shalousun Hello, I would like to ask when our version 3.0.5 will be released?

I tested it today, and the current Javadoc needs to be optimized. The definition of methods in the Javadoc needs to know the modifiers, such as the difference between static methods and ordinary public instance methods.

<img width="935" alt="image" src="https://github.com/TongchengOpenSource/smart-doc/assets/9592228/4c8c549b-579c-42d3-8d17-ca59707fd4e7">


After this optimization is completed, we can schedule a release for version 3.0.5 next week.

**Reply by tuichenchuxin** [reply]
@shalousun Ok, I'll try to fix it.

**Reply by shalousun** [reply]
I have already fixed it. #801 #802

**Reply by tuichenchuxin** [reply]
Wow, that's awesome!
